doc/developer-guides/coding_guidelines.rst

```diff
@@ -695,6 +724,299 @@ Compliant example::
        }
 
 
+FN-11: The return type of a function shall be consistent
+========================================================
+
+The return type of a function shall be the same between its declaration and
+definition.
+
+Compliant example::
+
+    /* In `showcase.h`: */
+    uint32_t func_showcase(uint32_t param);
+    
+    /* In `showcase.c`: */
+    #include "showcase.h"
+    
+    uint32_t func_showcase(uint32_t param)
+    {
+        return param;
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       /* In `showcase.h`: */
+       uint64_t func_showcase(uint64_t param);
+       
+       /* In `showcase.c`: */
+       #include "showcase.h"
+       
+       uint32_t func_showcase(uint32_t param)
+       {
+               return param;
+       }
+
+
+FN-12: Banned functions shall not be used
+=========================================
+
+The following cases shall be covered:
+
+a) These dynamic memory allocation functions shall not be used: ``calloc``,
+   ``malloc``, ``realloc``, and ``free``.  Dynamic memory allocation shall be
+   replaced with static memory allocation.
+b) The functions ``va_arg``, ``va_start``, and ``va_end`` shall only be used
+   within variadic functions (functions taking a variable number of parameters)
+   such as ``printf``.
+
+Compliant example::
+
+    uint32_t showcase_array[32];
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t *showcase_ptr = (uint32_t *)malloc(32U * sizeof(uint32_t));
+
+
+FN-13: All declared functions shall have a corresponding definition
+===================================================================
+
+Compliant example::
+
+    /* In `showcase.h`: */
+    /* declaration */
+    uint32_t func_showcase(uint32_t param);
+    
+    /* In `showcase.c`: */
+    #include "showcase.h"
+    
+    /* definition */
+    uint32_t func_showcase(uint32_t param)
+    {
+        return param;
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       /* In `showcase.h`: */
+       /* declaration */
+       uint32_t func_showcase(uint32_t param);
+       
+       /* There is no definition of `func_showcase` anywhere in all source files */
+
+
+FN-14: All defined functions shall be used
+==========================================
+
+All defined functions shall be used, either called explicitly or indirectly via
+its address. Otherwise, the function shall be removed. The following case is an
+exception. Some extra functions may be kept in order to provide a more complete
+library of APIs. These functions may have been implemented but not used
+currently. These functions will come in handy in the future. In this case, these
+functions may not be removed.
```

Comment:
This wording prohibits removing these "library" functions when your intention is to say they may remain:
```suggestion
functions may remain.
```